src/Voron/Util/ThreadHoppingReaderWriterLock.cs

```diff
@@ -90,28 +90,38 @@ private bool TryEnterReadLockSlow(int timeout)
             {
                 bool lockTaken = false;
                 _readWaitLock.TryEnter(tracker.RemainingMilliseconds, ref lockTaken);
+
                 try
                 {
                     if (lockTaken == false)
                         return false;
 
+                    ForTestingPurposes?.BeforeResetOfReaderWait?.Invoke();
+
                     if (_readerWait.IsSet)
                         _readerWait.Reset();
 
                     if (TryEnterReadLockCore())
+                    {
+                        // we got the reader lock, we can release all the other readers that might be waiting
+                        _readerWait.Set();
```

Comment:
Technically this can / should be set only if we called `_readerWait.Reset();` above. Only then we have this issue. Not sure if it's worth to introduce a boolean flag for that to detect that we called reset.

Reply:
It could be more clear and self explaining

Reply:
changed.
If the `_readerWait` is set, it means that we can `TryEnterReadLockCore`.

Reply:
Talked to @ayende. Decided not to add a boolean flag.